### PR TITLE
fix(core): bound DOCX table extraction

### PR DESCRIPTION
**Pull request by jan-kubica**
### Motivation

- The previous table-to-markdown extraction capped columns but not row count or total emitted characters, allowing attacker-controlled DOCX tables to expand into very large markdown output and exhaust CPU/memory. 

### Description

- Add `MAX_TABLE_CHARACTERS` and `MAX_TABLE_ROWS` limits and a `TableExtractionBudget` to bound per-extraction table markdown expansion. 
- Enforce the row limit in `readTableGrid` and enforce the cumulative character budget in `renderTableRows`, raising a typed `DocxArchiveError` on overflow. 
- Thread the shared `tableBudget` through `extractParts`, `extractContainer`, and table rendering so headers, body, and footers cannot bypass the global limit. 
- Add a regression test that constructs a wide table with many empty rows and asserts the extractor rejects it, and add a changeset (`.changeset/limit-docx-table-extraction.md`).

### Testing

- `git diff --check` completed with no issues. 
- `bun build` of the patched module completed successfully (the extractor bundles). 
- `bun test` was attempted but could not complete because dependency installation failed (registry returned HTTP 403 and `jszip` was unavailable), so the new test could not be executed in this environment. 
- Formatting/linting and `bun install` could not be run to completion for the same dependency-registry reasons.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a765a641094833384d12f69e7a5db2c)

## Summary by CodeRabbit

* **Bug Fixes**
  * Added safeguards when extracting DOCX tables to prevent oversized or excessively padded content from consuming excessive resources.
  * DOCX extraction now rejects documents with more than 10,000 table rows or 8 MiB of rendered table content.
  * Safeguards apply consistently across document bodies, headers and footers.

### CODE REVIEW

File: packages/core/src/docx/server/extractDocxText.ts [hunk 1]
```diff
@@ -327,6 +332,12 @@ const readTableGrid = (table: XmlElement): TableGrid => {
   let firstRowIsHeader = false;
 
   for (const [rowIndex, row] of collectTableParts(table, "tr").entries()) {
+    if (rowIndex >= MAX_TABLE_ROWS) {
+      throw new DocxArchiveError({
+        message: `Extracted DOCX table exceeded the ${MAX_TABLE_ROWS}-row limit`,
+        reason: "total-too-large",
+      });
+    }
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/core/src/docx/server/extractDocxText.ts]: Implemented in 6c657edd. Table rows and cells now reject at the collection boundary before an over-limit element is retained; source characters are charged while cell content is expanded, and rendered characters are charged before rows enter the output array. Added regressions for both row-count and character-total exhaustion. CC on behalf of sok0

**Comment by jan-kubica** [comment]
Addressed the remaining review items in 6c657edd: `renderTableRows` now takes named options, the archive limit reason is a named constant, table part collection rejects before retaining over-limit rows or cells, and separate source/rendered character budgets are enforced before content is retained. Focused extraction tests, repository lint, format check, and full typecheck pass; the complete test suite is still running alongside fresh CI. CC on behalf of sok0

**Comment by jan-kubica** [comment]
Rebased without semantic changes onto current `main`; exact head is now 193cb115. The complete local test suite also passed, alongside lint, format, and typecheck. CC on behalf of sok0
